File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        total_product = 1
        ans = []
        k = 0
        for num in nums:
            if num != 0:
                total_product *= num
            else:
                k += 1

        for num in nums:
            if num != 0:
                if k > 0:
                    ans.append(0)
                else:
                    ans.append(total_product // num)
            elif num == 0 and k > 1:
                ans.append(0)
            else:
                ans.append(total_product)
        return ans

File: Data Structures & Algorithms/majority-element/submission-3.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        if (nums.size() == 1) return nums[0];

        int count = 1;
        sort(nums.begin(), nums.end());
        int ans = nums[0];
        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums[i] == nums[i + 1]) {
                count++;
                if (count > nums.size() / 2) {
                    ans = nums[i];
                    break; 
                }
            } else {
                count = 1;
            }
        }
        return ans;
    }
};